azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -70,24 +77,90 @@
 
     @Override
     protected void doExecute() throws AzureExecutionException {
-        try {
-            final RuntimeHandler runtimeHandler = getFactory().getRuntimeHandler(
-                    getWebAppConfiguration(), getAzureClient());
-            // todo: use parser to get web app from mojo configs
-            final WebApp app = getWebApp();
-            if (app == null) {
-                if (this.isDeployToDeploymentSlot()) {
-                    throw new AzureExecutionException(WEBAPP_NOT_EXIST_FOR_SLOT);
-                }
-                createWebApp(runtimeHandler);
-            } else {
-                updateWebApp(runtimeHandler, app);
-            }
-            deployArtifacts(getWebAppConfiguration());
-        } catch (IOException | AzureAuthFailureException | InterruptedException e) {
-            throw new AzureExecutionException(
-                    String.format("Encountering error when deploying to azure: '%s'", e.getMessage()), e);
+        final WebAppConfig config = getWebAppConfig();
+        final IAppService target = createOrUpdateResource(config);
+        deployArtifacts(target, config);
+    }
+
+    private IAppService createOrUpdateResource(final WebAppConfig config) throws AzureExecutionException {
+        final IAppService target = getDeployTarget(config);
```

Comment:
getDeployTarget is confusing, given the name, we cannot expect the result, consider a help method of isDeploymentSlot